merkle/map_verifier.go

```diff
@@ -40,9 +40,6 @@ func VerifyMapInclusionProof(index []byte, leafHash []byte, expectedRoot []byte,
 	if got, want := len(leafHash)*8, hBits; got != want {
 		return fmt.Errorf("invalid leafHash length %d, expected %d", got, want)
 	}
-	if got, want := len(expectedRoot)*8, hBits; got != want {
```

Comment:
By "clean" I mean that tests can use `[]byte("fake root")` instead of `hasher.LeafHash([]byte("fake root")`  Requiring test values to be the output of a hash function doesn't make much sense, especially when we're trying to test for error cases.